src/checker/hunspell.rs

```diff
@@ -25,12 +25,23 @@ use anyhow::{anyhow, bail, Result};
 use super::quirks::{
     replacements_contain_dashed, replacements_contain_dashless, transform, Transformed,
 };
+use regex::RegexSetBuilder;
 
 pub struct HunspellWrapper(pub Arc<Hunspell>);
 
 unsafe impl Send for HunspellWrapper {}
 unsafe impl Sync for HunspellWrapper {}
 
+// The value is `true` if string is made of emoji's
+// or unicode `VULGAR FRACTION`
+pub fn vulgar_fraction_or_emoji(word: &str) -> bool {
+    let set = RegexSetBuilder::new(&[r"[\u00BC-\u00BE\u2150-\u215E-\u2189]", r"^[\p{Emoji}]+$"])
```

Comment:
Compiling regexp is quite expensive, so this should definitely be called in a lazy static at a very very early point (not that since this might be called from a thread pool with rayon, make sure this is initialized early on in `Hunspell::check()`